File: Fibonacci Heaps/FibonacciIncreaseKey.java
// File: FibonacciIncreaseKey.java
// Program: Increase a key value in a Fibonacci Heap

class Node {
    int key;
    Node left, right;

    Node(int key) {
        this.key = key;
        left = right = this;
    }
}

public class FibonacciIncreaseKey {

    static void increaseKey(Node x, int newKey)
    {
        if (newKey < x.key)
        {
            System.out.println("New key is smaller than current key");
            return;
        }

        x.key = newKey;
        System.out.println("Key increased to " + newKey);
    }

    public static void main(String[] args)
    {
        Node node = new Node(10);

        increaseKey(node, 25);
    }
}

File: Fibonacci Heaps/fibonacci_increase_key.c
/* File: fibonacci_increase_key.c
   Program: Increase a key value in a Fibonacci Heap
*/

#include <stdio.h>
#include <stdlib.h>

typedef struct node {
    int key;
    struct node *left;
    struct node *right;
} Node;

Node *min = NULL;

Node* createNode(int key)
{
    Node *n = (Node*)malloc(sizeof(Node));
    n->key = key;
    n->left = n;
    n->right = n;
    return n;
}

Node* insert(Node *min, int key)
{
    Node *n = createNode(key);

    if (min == NULL)
        return n;

    n->left = min;
    n->right = min->right;
    min->right->left = n;
    min->right = n;

    if (n->key < min->key)
        min = n;

    return min;
}

void increaseKey(Node *x, int newKey)
{
    if (newKey < x->key)
    {
        printf("New key is smaller than current key\n");
        return;
    }

    x->key = newKey;
    printf("Key increased to %d\n", newKey);
}

int main()
{
    Node *n1 = createNode(10);

    min = insert(min, 10);

    increaseKey(n1, 25);

    return 0;
}
